### ISSUE TITLE
real reference for API

### ISSUE THREAD

**Issue by ShashankSureshRotti** [prometheus/client_python#1021]
> https://prometheus.github.io/client_python/

this is like a getting start hello world example

is there like a real API ref for the classes etc..

**Comment by kakkoyun** [comment]
Thanks for reaching out. 

That's true; we can add more complicated real-life examples. However, the package is also used by a significant number of open-source Python applications that are publicly available.

Please let us know if you have anything particular you would like to see in the docs.

**Comment by Azmisov** [comment]
Yes, I would like to see API docs as well. Examples are fine, but you need to document the API. For example, creating a `Gauge` instance: the example given under the Gauge page doesn't document all the parameters that can be passed to the Gauge constructor, nor supplementary methods you can call like `.labels`.

**Comment by Digicrat** [comment]
Another vote for publishing a full API documentation (even just an auto-generated one).  

More simply, I also suggest adding to the main "Getting Started" page links to a handful of those publicly available open-source applications using this API as real-world examples.  The main Prometheus documentation includes links to many examples, but does not differentiate what language they are written in (most are in Go), making it difficult to find relevant Python-based examples.

**Comment by funkyfuture** [comment]
my current use case would be that i'm looking at scripts here: https://github.com/prometheus-community/node-exporter-textfile-collector-scripts and they use functions and classes that are not mentioned in the examples. a complete inventory of the API would enable me to find what i need to solve my problems w/o reading the source code.

**Comment by csmarchbanks** [comment]
That sounds great! The first two are the most important for this library, if anything for the explanations of which metric types/getting started pointing to the[ core prometheus docs](https://prometheus.io/docs/concepts/metric_types/) would make sense.

Thanks!

**Comment by k1chik** [comment]
Hi @csmarchbanks, I'd like to pick this up! I see there was interest earlier, but no PR has been opened. I can start with the structured API reference for Counter, Gauge, Histogram, and Summary with real-world usage examples. Happy to coordinate if someone else is actively working on it :)

**Comment by csmarchbanks** [comment]
:+1: I don't know of anyone else working on it (haven't seen anything from the above person) so please feel free to start!

**Comment by k1chik** [comment]
Hey @csmarchbanks, made the PR and it's waiting for review :)
